components/FavoriteContactsList.js

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import { fetchContacts } from "../actions/contacts";
+import { getContacts } from "../selectors/contacts";
+
+import ContactItem from "./ContactItem";
+
+export default function FavoriteContactList({ favorite }) {
+  const dispatch = useDispatch();
+  const contacts = useSelector(getContacts);
+
+  useEffect(() => {
+    dispatch(fetchContacts());
+  }, []);
```

Comment:
hmm, looks like you'd be fetching contacts twice? once in contacts list and once in favorites list

Reply:
true - i guess in the case that it was rendered separately and before the contacts list in a hypothetical scenario it might be worth keeping, but for the scope of this app, definitely not necessary. guess i could add a quick check to state before this to reduce calls if that were the case. good catch!